hibernate-core/src/main/java/org/hibernate/Hibernate.java

```diff
@@ -88,6 +86,9 @@ public static void initialize(Object proxy) throws HibernateException {
 		else if ( proxy instanceof PersistentCollection ) {
 			( (PersistentCollection<?>) proxy ).forceInitialization();
 		}
+		else if ( proxy instanceof LazyInitializable ) {
+			( (LazyInitializable) proxy ).forceInitialization();
+		}
```

Comment:
You don't need two `if` branches, as `PersistentCollection` extends `LazyInitializable`:

```suggestion
		else if ( proxy instanceof LazyInitializable ) {
			( (LazyInitializable) proxy ).forceInitialization();
		}
```

Reply:
I thought for a moment that leaving the two if's might simplify the understanding for everyone who is used to working with PersistenCollection. I agree that it is correct to eliminate double checking. I will make the change.